fem/kdtree.cpp

```diff
@@ -176,8 +176,8 @@ void KDTreeNodalProjection<2>::Project(const GridFunction& gf, real_t lerr)
       }
 
       // initialize bbmax and bbmin
-      const FiniteElement* el=gf.FESpace()->GetFE(0);
-      trans = gf.FESpace()->GetElementTransformation(0);
+      const FiniteElement* el=gf.FESpace()->GetTypicalFE();
+      trans = gf.FESpace()->GetMesh()->GetTypicalElementTransformation();
```

Comment:
First, I don't think this code is meant to run on the local portion of a parallel mesh, so assuming we have elements is probably just fine. @bslazarov, is this correct?

If this is supposed to work for empty local mesh (maybe not now but in the future), I'm not sure using some dummy `ElementTransformation` is the right thing to use here (if `Mesh::GetTypicalElementTransformation()` returns a dummy `ElementTransformation` in this case). I suspect that this case needs to be handled explicitly here.

Reply:
I think you're right, reverted in 06f60e28d.